tools/dbbuilder/dbbuilder.cpp

```diff
@@ -189,26 +189,67 @@ int main(int argc, char* argv[])
     if (access(logFile.c_str(), W_OK) != 0)
       canWrite = false;
 
+    // MCOL-5162 Automatic syscat upgrade
+    // The below std::unordered_map's are used by the
+    // SystemCatalog::upgrade() call.
+    bool isUpgrade = false;
+
+    std::unordered_map<int, std::pair<int, bool>> upgradeOidMap;
+//    upgradeOidMap[OID_SYSTABLE_AUXCOLUMNOID] =      // This is the candidate OID for the upgrade.
+//      std::make_pair(OID_SYSTABLE_OBJECTID, false); // std::pair::first is the reference OID used
+                                                    // to fill the candidate OID with default vals
+                                                    // std::pair::second is set to false by default
+                                                    // which means the candidate OID will not be
+                                                    // upgraded. This is set to true in the code
+                                                    // below if a specific condition is met. Please
+                                                    // note that the candidate and reference OID
+                                                    // datatypes and colwidths are assumed to be the
+                                                    // same in SystemCatalog::upgrade().
+    upgradeOidMap[OID_SYSCOLUMN_COMPRESSIONTYPE] =
+      std::make_pair(OID_SYSCOLUMN_OBJECTID, false);
+
+    std::unordered_map<int, std::pair<CalpontSystemCatalog::ColDataType, int>> upgradeOidTypeMap;
+//    upgradeOidTypeMap[OID_SYSTABLE_AUXCOLUMNOID] =
+//      std::make_pair(CalpontSystemCatalog::INT, 4);
+    upgradeOidTypeMap[OID_SYSCOLUMN_COMPRESSIONTYPE] =
+      std::make_pair(CalpontSystemCatalog::INT, 4);
+
+    std::unordered_map<int, std::string> upgradeOidDefaultValStrMap;
+//    upgradeOidDefaultValStrMap[OID_SYSTABLE_AUXCOLUMNOID] = "0";
+    upgradeOidDefaultValStrMap[OID_SYSCOLUMN_COMPRESSIONTYPE] = "0";
+
```

Comment:
@benthompson15 
To add support for the `collation` column, the change is as simple as adding an entry to the above three unordered_map's. Read the above comment as well as the commit description. I added the `OID_SYSCOLUMN_COMPRESSIONTYPE` entry just as a demonstration, this should be replaced with the entry for the `collation` columns in `systable` and `syscolumn`.